ast/compile.go

```diff
@@ -4761,7 +4761,12 @@ func checkUnusedAssignedVars(loc *Location, stack *localDeclaredVars, used VarSe
 	unused = unused.Diff(rewrittenUsed)
 
 	for _, gv := range unused.Sorted() {
-		errs = append(errs, NewError(CompileErr, loc, "assigned var %v unused", dvs.reverse[gv]))
+		for i := range body {
+			if body[i].Vars(VarVisitorParams{}).Contains(gv) {
+				// TODO: better way to get location of assignment?
```

Comment:
I think we're good 👍. There is precedence for doing it like this in `checkUnusedDeclaredVars()`, and since this is only executed in an error scenario, performance isn't our greatest concern.